docs/latest/design/request-authentication.md

```diff
@@ -0,0 +1,507 @@
+    - matches:
+        - path:
+            type: PathPrefix
+            value: /foo
+      filters:
+        - type: ExtensionRef
+          extensionRef:
+            group: security.gateway.envoyproxy.io
+            kind: Authentication
+            name: example1
+      backendRefs:
+        - name: backend
+          port: 3000
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: example2
+spec:
+  hostnames:
+    - www.example2.com
+  parentRefs:
+    - group: gateway.networking.k8s.io
+      kind: Gateway
+      name: eg
+  rules:
+    - matches:
+        - path:
+            type: PathPrefix
+            value: /bar
+      filters:
+        - type: ExtensionRef
+          extensionRef:
+            group: security.gateway.envoyproxy.io
+            kind: Authentication
+            name: example2
+      backendRefs:
+        - name: backend2
+          port: 3000
+```
+
+The following xDS configuration is created from the above example resources:
+
+```yaml
+configs:
+...
+dynamic_listeners:
+  - name: default-eg-http
+    ...
+        default_filter_chain:
+          filters:
+          - name: envoy.filters.network.http_connection_manager
+            typed_config:
+              '@type': >-
+                type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+              stat_prefix: http
+              rds:
+                config_source:
+                  ...
+                route_config_name: default-eg-http
+              http_filters:
+                - name: envoy.filters.http.jwt_authn
+                  typed_config:
+                    '@type': >-
+                      type.googleapis.com/envoy.config.filter.http.jwt_authn.v2alpha.JwtAuthentication
+                    providers:
+                      default-example1-example1:
+                        issuer: https://www.example1.com
+                        audiences:
+                          - foo.com
+                        remote_jwks:
+                          http_uri:
+                            uri: https://foo.com/jwt/public-key/jwks.json
+                            cluster: default-example1-example1-jwt
+                      default-example2-example2:
+                        issuer: https://www.example2.com
+                        audiences:
+                          - bar.com
+                        remote_jwks:
+                          http_uri:
+                            uri: https://bar.com/jwt/public-key/jwks.json
+                            cluster: default-example2-example2-jwt
+                    rules:
+                      - match:
+                          exact: /foo
+                        requires:
+                          provider_name: default-example1-example1
+                      - match:
+                          exact: /bar
+                        requires:
+                          provider_name: default-example2-example2
```

Comment:
This works, though we might end up with [per route config](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/jwt_authn/v3/config.proto#extensions-filters-http-jwt-authn-v3-perrouteconfig) and requirement map when this grows.